pallets/creator-staking/src/lib.rs

```diff
@@ -0,0 +1,748 @@
+                let next_era = previous_era + 1;
+                CurrentEra::<T>::put(next_era);
+
+                NextEraStartingBlock::<T>::put(now + blocks_per_era);
+
+                let reward = BlockRewardAccumulator::<T>::take();
+                Self::reward_balance_snapshot(previous_era, reward);
+                let consumed_weight = Self::rotate_staking_info(previous_era);
+
+                if force_new_era {
+                    ForceEra::<T>::put(Forcing::NotForcing);
+                }
+
+                Self::deposit_event(Event::<T>::NewCreatorStakingEra { number: next_era });
+
+                consumed_weight + T::DbWeight::get().reads_writes(5, 3)
+            } else {
+                T::DbWeight::get().reads(4)
+            }
+        }
+    }
+
+    #[pallet::call]
+    impl<T: Config> Pallet<T> {
+        #[pallet::call_index(0)]
+        #[pallet::weight(Weight::from_ref_time(100_000) + T::DbWeight::get().reads_writes(3, 1))]
+        pub fn force_register_creator(
+            origin: OriginFor<T>,
+            space_id: SpaceId,
+        ) -> DispatchResultWithPostInfo {
+            Self::ensure_pallet_enabled()?;
+            ensure_root(origin)?;
+
+            ensure!(
+                !RegisteredCreators::<T>::contains_key(space_id),
+                Error::<T>::AlreadyUsedCreatorSpace,
+            );
+
+            let space_owner = T::SpacesInterface::get_space_owner(space_id)?;
+            T::Currency::reserve(&space_owner, T::RegistrationDeposit::get())?;
+
+            RegisteredCreators::<T>::insert(space_id, CreatorInfo::new(space_owner.clone()));
+
+            Self::deposit_event(Event::<T>::CreatorRegistered { who: space_owner, space_id });
+
+            Ok(().into())
+        }
+
+        #[pallet::call_index(1)]
+        #[pallet::weight(Weight::from_ref_time(100_000) + T::DbWeight::get().reads_writes(2, 1))]
+        pub fn unregister_creator(
+            origin: OriginFor<T>,
+            space_id: SpaceId,
+        ) -> DispatchResultWithPostInfo {
+            let who = ensure_signed(origin)?;
+
+            Self::do_unregister_creator(space_id, UnregistrationAuthority::Creator(who.clone()))?;
+
+            Self::deposit_event(Event::<T>::CreatorUnregistered { who, space_id });
+
+            Ok(().into())
+        }
+
+        #[pallet::call_index(2)]
+        #[pallet::weight(Weight::from_ref_time(100_000) + T::DbWeight::get().reads_writes(2, 1))]
+        pub fn force_unregister_creator(
+            origin: OriginFor<T>,
+            space_id: SpaceId,
+        ) -> DispatchResultWithPostInfo {
+            ensure_root(origin)?;
+
+            Self::do_unregister_creator(space_id, UnregistrationAuthority::Root)?;
+            let creator_info =
+                Self::registered_creator(space_id).ok_or(Error::<T>::NotRegisteredCreator)?;
+
+            Self::deposit_event(Event::<T>::CreatorUnregisteredWithSlash {
+                who: creator_info.stakeholder,
```

Comment:
But `who` should be `root` address. Because usually we put `origin` into `who` field of events.

Reply:
We can't without tight coupling pallet_sudo to this pallet. As far as `who` will always be root here, it's redundant to pass it to the event.